### PR TITLE
feat: Add a cache for arbitrary unvalidated downloads

### PR DESCRIPTION
**Pull request by Swatinem**
This might be useful for downloading individual source files.

#skip-changelog

### CODE REVIEW

File: crates/symbolicator-service/src/services/caches/downloads.rs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+use super::versions::DOWNLOADS_CACHE_VERSIONS;
+
+/// Provides cached access to arbitrary downloaded files, without validating or parsing their contents.
+///
+/// This should be used for any files which are not being validated / converted upon download.
+/// An example might be downloads for individual source files.
+///
+// XXX: The only "validation" that source files might need is checking that they are valid UTF-8.
+// Not sure if its worth validating this at download time, to persist a `Malformed` marker?
```
Comment (Swatinem) [inline crates/symbolicator-service/src/services/caches/downloads.rs:24]: This might be a valid concern. Not sure if this cache might be a bit too generic?

**Review by loewenheim — APPROVED** [review]

File: crates/symbolicator-service/src/services/caches/downloads.rs (hunk 1 above)
Comment (loewenheim) [reply crates/symbolicator-service/src/services/caches/downloads.rs:24]: I think it's probably reasonable to do UTF-8 validation here; I don't think we anticipate this being used for arbitrary binary files?

Comment (kamilogorek) [reply crates/symbolicator-service/src/services/caches/downloads.rs:24]: I'm not sure if it's worth the time to do the validation, but I think either choice are fine.

**Review by kamilogorek — APPROVED** [review]

**Comment by Swatinem** [comment]
After some thinking, maybe it would be good to still do utf-8 validation at download and load-time, and expose this maybe as a `SelfCell<&'static str, ByteView<'static>>` or whatever we need to do to have a validated mmapped `&str`.
Renaming this from a generic `"download"` cache to maybe `"sourcefile"` or something that makes it clear that we are using this as utf-8 validated source files might make sense.

WDYT? CC @vaind

**Comment by loewenheim** [comment]
>  Renaming this from a generic "download" cache to maybe "sourcefile" or something that makes it clear that we are using this as utf-8 validated source files might make sense.

I agree that this name would fit the use case of "definitely utf-8" files better.

**Review by loewenheim — APPROVED** [review]

**Pull request merged**